src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -1913,7 +1929,26 @@ void Creature::Respawn(bool force)
             // Respawn check if spawn has 2 entries
             if (data->id2)
             {
-                uint32 entry = (roll_chance_f(data->chance_id1)) ? data->id : data->id2;
+                // Add to world
+                uint32 entry = data->id;
+                uint8 ids = 1;
+                if (data->id3) ++ids;
+
+                if (ids)
+                {
+                    uint8 idNumber = urand(0, ids);
+                    switch (idNumber) {
```

Comment:
Make this a function so ypu dont have to use the exact same code twice

Reply:
Creature::GetRandomId() or something

Reply:
Just a mock up I thought of
```cpp
entry = GetRandomId(id2, id3);
```
```cpp
Creature::GetRandomId(bool hasId2, bool hasId3)
{
    if (!hasId2)
        return id = data->id1;

    unit8 ids = 0;
    if (hasId2) ++ids;
    if (hasId3) ++ids;

    if (ids)
    {
        uint8 idNumber = urand(0, ids);
        switch (idNumber)
            case 0:
                id = data->id1;
                break;
            case 1:
                id = data->id2;
                break;
            case 2:
                id = data->id3;
                break;
    }

    return id;
}
```

Reply:
Updated. Will push after compile.